docs/reference/data-streams/set-up-a-data-stream.asciidoc

```diff
@@ -54,139 +87,145 @@ PUT /_ilm/policy/my-data-stream-policy
+        "@timestamp": {
+          "type": "date",
+          "format": "date_optional_time||epoch_millis"
+        },
+        "message": {
+          "type": "wildcard"
+        }
+      }
+    }
+  }
+}
 
-The naming scheme separates these components with a `-` character:
+# Creates a component template for index settings
+PUT _component_template/my-settings
+{
+  "template": {
+    "settings": {
+      "index.lifecycle.name": "my-lifecycle-policy"
+    }
+  }
+}
+----
+// TEST[continued]
 
-```
-<type>-<dataset>-<namespace>
-```
+[discrete]
+[[create-index-template]]
+=== Step 3. Create an index template
 
-For example, the {agent} uses the `logs-nginx.access-production` data
-stream to store data with a type of `logs`, a dataset of `nginx.access`, and a
-namespace of `production`. If you use the {agent} to ingest a log file, it
-stores the data in the `logs-generic-default` data stream.
+Use your component templates to create an index template. Specify:
 
-For more information about the naming scheme and its benefits, see our
-https://www.elastic.co/blog/an-introduction-to-the-elastic-data-stream-naming-scheme[An
-introduction to the Elastic data stream naming scheme] blog post.
-****
+* One or more index patterns that match the data stream's name. We recommend
+using {fleet-guide}/data-streams.html#data-streams-naming-scheme[our data
+stream naming scheme].
 
-include::{es-repo-dir}/data-streams/data-streams.asciidoc[tag=timestamp-reqs]
+* That the template is data stream enabled.
 
-If using {ilm-init}, specify your lifecycle policy in the `index.lifecycle.name`
-setting.
+* Any component templates that contain your mappings and index settings.
 
-TIP: Carefully consider your template's mappings and settings. Later changes may
-require reindexing. See <<data-streams-change-mappings-and-settings>>.
+* A priority higher than `200` to avoid collisions with built-in templates.
+See <<avoid-index-pattern-collisions>>.
 
-[role="screenshot"]
-image::images/data-streams/create-index-template.png[Create template page]
+To create an index template in {kib}, open the main menu and go to *Stack
+Management > Index Management*. In the *Index Templates* view, click *Create
+template*.
 
-[%collapsible]
-.API example
-====
-Use the <<indices-put-template,create or update index template API>> to create
-an index template. The template must include a `data_stream` object, indicating
-it's used for data streams.
+You can also use the <<indices-put-template,create index template API>>.
+Include the `data_stream` object to enable data streams.
 
 [source,console]
 ----
-PUT /_index_template/my-data-stream-template
+PUT _index_template/my-index-template
 {
-  "index_patterns": [ "my-data-stream*" ],
+  "index_patterns": ["my-data-stream*"],
   "data_stream": { },
-  "priority": 500,
-  "template": {
-    "settings": {
-      "index.lifecycle.name": "my-data-stream-policy"
-    }
-  }
+  "composed_of": [ "my-mappings", "my-settings" ],
+  "priority": 500
```

Comment:
Do you think it's worth adding an example `_meta` object with a description, to show a user how arbitrary metadata can be added to component and index templates?